shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java

```diff
@@ -161,30 +161,31 @@
 	ResponseEntity<Boolean> hasOneStudyToImport() throws RestServiceException;
 
 	@ApiOperation(value = "", notes = "Add protocol file to a study", response = Void.class, tags = {})
-	@ApiResponses(value = { @ApiResponse(code = 204, message = "examination updated", response = Void.class),
+	@ApiResponses(value = {
+	        @ApiResponse(code = 200, message = "protocol file", response = Void.class),
 			@ApiResponse(code = 401, message = "unauthorized", response = Void.class),
 			@ApiResponse(code = 403, message = "forbidden", response = Void.class),
 			@ApiResponse(code = 422, message = "bad parameters", response = ErrorModel.class),
 			@ApiResponse(code = 500, message = "unexpected error", response = ErrorModel.class) })
 	@PostMapping(value = "protocol-file-upload/{studyId}", produces = { "application/json" }, consumes = {
 			"multipart/form-data" })
-	@PreAuthorize("hasRole('ADMIN')")
+	@PreAuthorize("hasAnyRole('ADMIN', 'EXPERT') and @studySecurityService.hasRightOnStudy(#studyId, 'CAN_ADMINISTRATE')")
 	ResponseEntity<Void> uploadProtocolFile(
 			@ApiParam(value = "id of the study", required = true) @PathVariable("studyId") Long studyId,
 			@ApiParam(value = "file to upload", required = true) @Valid @RequestBody MultipartFile file)
 			throws RestServiceException;
 
 	@ApiOperation(value = "", notes = "Download protocol file from a study", tags = {})
 	@ApiResponses(value = {
-			@ApiResponse(code = 204, message = "examination updated"),
+	        @ApiResponse(code = 200, message = "protocol file", response = Resource.class),
 			@ApiResponse(code = 401, message = "unauthorized", response = Void.class),
 			@ApiResponse(code = 403, message = "forbidden", response = Void.class),
 			@ApiResponse(code = 422, message = "bad parameters", response = ErrorModel.class),
 			@ApiResponse(code = 500, message = "unexpected error", response = ErrorModel.class) })
 	@GetMapping(value = "protocol-file-download/{studyId}/{fileName:.+}/")
-	@PreAuthorize("hasRole('ADMIN')")
+	@PreAuthorize("hasRole('ADMIN') or (hasAnyRole('EXPERT', 'USER') and @studySecurityService.hasRightOnStudy(#studyId, 'CAN_SEE_ALL'))")
```

Comment:
Hi Jean-Come, could you please change this to can_download? every user has can_see_all from default, when member of the study. From my point of view the idea is, that with can_see users can look on everything on the web, but can not download/"take away" contents. I hope this sounds logic. With kind regards, Michael